### ISSUE TITLE
loadGTFS doesn't work.

### ISSUE THREAD

**Issue by HTVsuper** [planarnetwork/raptor#17]
So I've tried the readme examples, but it gets stuck on loadGTFS. The gtfs file I am using is this [one]( https://rata.digitraffic.fi/api/v1/trains/gtfs-passenger.zip) from https://www.digitraffic.fi/en/railway-traffic/#gtfs.

**Comment by linusnorton** [comment]
I fairly quick seems to indicate it's working with that file:

HKI_7 -> LR_1 today at 6am shows:

06:55:00, 09:07:00, HKI_7-LR_1

What inputs are you using and what output are you getting?

**Comment by HTVsuper** [comment]
Oh ok then it's probably a user issue, since I'm quite inexperienced with JS.

I didn't even find where to put any inputs.

**Comment by piemadd** [comment]
I am having the same problem. Combing through the code and adding comments at key points, I seem to have narrowed down the issue to something with the `gtfs-stream` library not sending an `end`. 

![image](https://github.com/planarnetwork/raptor/assets/22846928/963f9f95-f6f9-4c2f-ac47-1a44a9142f0a)
![image](https://github.com/planarnetwork/raptor/assets/22846928/e544ae76-3f01-439d-a5d1-e57e0dfc9161)
![image](https://github.com/planarnetwork/raptor/assets/22846928/8d04c4f3-d3cc-4de9-befe-4bca3264177d)

**Comment by piemadd** [comment]
Whoops, forgot to add this. The loader gets initialized, but never seems to resolve the promise.

![image](https://github.com/planarnetwork/raptor/assets/22846928/d3b4c7f4-b019-47eb-93df-23cd7890377c)

**Comment by piemadd** [comment]
With even more testing, this is definitely a bug with `gtfs-stream`. Tried doing a test with this script:
```js
const fs = require('fs');
const gtfs = require('gtfs-stream')

fs.createReadStream("gtfs.zip")
  .pipe(gtfs())
  .on('data', (entity) => {
    //console.log(entity)
  })
  .on('end', () => {
    console.log('end')
  })
```
And the program is exited without 'end' ever being logged. Heading over to that repo to file an issue now.

**Comment by linusnorton** [comment]
I have the same issue on https://github.com/planarnetwork/gtfsmerge/issues/11 - can you raise a reproducible error on gtfs-stream?

**Comment by piemadd** [comment]
Doing some more poking around, it seems like `gtfs-stream` sends a `finish` event instead of an `end` event (what raptor is expecting). Doing this allowed the gtfs to load, but then immediately popped up another bug for me 🙃.

**Comment by linusnorton** [comment]
There is a difference between finish and end but I can never remember what it is :) 

It was definitely working with "end" so I wonder if there is an error that's not being surfaced and then it's closing with "finish".

**Comment by piemadd** [comment]
Yeah that's true. Lemme add in a listener for errors really quick then.

**Comment by piemadd** [comment]
Hmmmm, no errors seem to be passed down. Something is going wrong somewhere but nothing is logging that.

**Comment by kostnerek** [comment]
Any clue what's wrong in RaptorAlgorithmFactory.create()? I'm coming to same conclusion as @piemadd, any tips how to fix this?

**Comment by matjamr** [comment]
Due to my observations calendars are not correctly processed, all calendar data is parsed into dates, hence broken matching for services, which results in further abominations. I will submit PR soon.

**Comment by matjamr** [comment]
> Submitted an issue to gtfs-stream [staeco/gtfs-stream#7](https://github.com/staeco/gtfs-stream/issues/7)
> 
> Now, there might be further issues with this, as poking through the code, it seems like the 4th item passed to `RaptorAlgorithmFactory.create()` is meant to be a `Date()` object, not the `calendars`.

It is mainly optional, which I assume, meant to accelerate search speed.

**Comment by kostnerek** [comment]
Wow nice @matjamr thank you, want to collaborate on fork/pr to resolve this issue?

**Comment by kostnerek** [comment]
Me and @kishieel are currently working on this maybe you want to join?
